File: backend/routes/game.py
"""
game.py – FastAPI router for /api/game endpoints.
"""

from typing import Any

from fastapi import APIRouter, Body
from fastapi.responses import JSONResponse

from controllers.gameController import (
    getStateHandler,
    initGame,
    legalMovesHandler,
    makeMoveHandler,
)

router = APIRouter()


@router.post("/init")
async def init_game_route(_payload: dict[str, Any] | None = Body(default=None)):
    result = await initGame()
    if isinstance(result, tuple):
        body, status = result
        return JSONResponse(status_code=status, content=body)
    return result


@router.post("/legal-moves")
async def legal_moves_route(payload: dict[str, Any] | None = Body(default=None)):
    result = await legalMovesHandler(payload)
    if isinstance(result, tuple):
        body, status = result
        return JSONResponse(status_code=status, content=body)
    return result


@router.post("/move")
async def move_route(payload: dict[str, Any] | None = Body(default=None)):
    result = await makeMoveHandler(payload)
    if isinstance(result, tuple):
        body, status = result
        return JSONResponse(status_code=status, content=body)
    return result


@router.get("/state")
async def state_route():
    result = await getStateHandler()
    if isinstance(result, tuple):
        body, status = result
        return JSONResponse(status_code=status, content=body)
    return result

File: backend/routes/game.js
/**
 * game.js – Express router for /api/game endpoints.
 *
 * Mounted in server.js as:   app.use('/api/game', gameRoutes)
 *
 * Endpoints:
 *   POST /api/game/init          → initGame
 *   POST /api/game/legal-moves   → legalMovesHandler
 *   POST /api/game/move          → makeMoveHandler
 *   GET  /api/game/state         → getStateHandler
 */

import express from 'express';
import {
  initGame,
  legalMovesHandler,
  makeMoveHandler,
  getStateHandler,
} from '../controllers/gameController.js';

const router = express.Router();

router.post('/init',         initGame);
router.post('/legal-moves',  legalMovesHandler);
router.post('/move',         makeMoveHandler);
router.get('/state',         getStateHandler);

export default router;
